src/DittoMachine.sol

```diff
@@ -278,7 +279,7 @@ contract DittoMachine is ERC1155D, ERC721TokenReceiver, ERC1155TokenReceiver, Cl
      * @param protoId specifies the clone to be burned.
      * @dev will refund funds held in a position, subsidy will remain for sellers in the future.
      */
-    function dissolve(uint protoId, uint cloneId) external {
+    function dissolve(uint protoId, uint cloneId) external returns (bool) {
```

Comment:
solidity doesn't do EXTCODESIZE check if a function call returns something. So this saves gas for smart contract callers.